sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareAsyncClient.java

```diff
@@ -152,6 +153,57 @@ public ShareFileAsyncClient getFileClient(String filePath) {
             serviceVersion);
     }
 
+    /**
+     * Gets if the share this client represents exists in the cloud.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.ShareAsyncClient.exists}
+     *
+     * @return true if the share exists, false if it doesn't
+     */
+    public Mono<Boolean> exists() {
+        try {
+            return existsWithResponse().flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
```

Comment:
Shouldn't need to try catch here given the overload being called already does this.